saiplayer/SaiPlayer.cpp

```diff
@@ -1022,309 +1049,338 @@ std::vector<std::string> SaiPlayer::tokenize(
     return tokens;
 }
 
-sai_status_t SaiPlayer::handle_bulk_route(
+sai_status_t SaiPlayer::handle_bulk_entry(
         _In_ const std::vector<std::string> &object_ids,
+        _In_ sai_object_type_t object_type,
         _In_ sai_common_api_t api,
         _In_ const std::vector<std::shared_ptr<SaiAttributeList>> &attributes,
-        _In_ const std::vector<sai_status_t>& expectedStatuses)
+        _Out_ std::vector<sai_status_t>& statuses)
 {
     SWSS_LOG_ENTER();
 
-    std::vector<sai_route_entry_t> routes;
+    sai_bulk_op_error_mode_t mode = SAI_BULK_OP_ERROR_MODE_IGNORE_ERROR; // TODO we need to get that from recording
+
+    uint32_t object_count = (uint32_t) object_ids.size();
 
-    for (size_t i = 0; i < object_ids.size(); ++i)
+    sai_status_t status = SAI_STATUS_SUCCESS;
+
+    if (api == SAI_COMMON_API_BULK_CREATE)
     {
-        sai_route_entry_t route_entry;
-        sai_deserialize_route_entry(object_ids[i], route_entry);
+        std::vector<uint32_t> attr_counts(object_count);
+        std::vector<const sai_attribute_t*> attr_lists(object_count);
 
-        route_entry.vr_id = translate_local_to_redis(route_entry.vr_id);
+        for (uint32_t idx = 0; idx < object_count; idx++)
+        {
+            attr_counts[idx] = attributes[idx]->get_attr_count();
+            attr_lists[idx] = attributes[idx]->get_attr_list();
+        }
 
-        routes.push_back(route_entry);
+        switch (object_type)
+        {
+            case SAI_OBJECT_TYPE_ROUTE_ENTRY:
+            {
+                std::vector<sai_route_entry_t> entries(object_count);
 
-        SWSS_LOG_DEBUG("route: %s", object_ids[i].c_str());
-    }
+                for (size_t it = 0; it < object_count; it++)
+                {
+                    sai_deserialize_route_entry(object_ids[it], entries[it]);
 
-    std::vector<sai_status_t> statuses;
+                    entries[it].switch_id = translate_local_to_redis(entries[it].switch_id);
```

Comment:
as for translation oids, there is generic function that translates automatically all oids in entry structure, but it must be meta_key, and this is trade off here, whether we copy meta_key and then extract actual route/nat/fdb to feed SAI api, or we manually translate each oid :(

Reply:
as for me, more quickly its manually translate each oid